File: Java/14.java
class Solution {
  public String longestCommonPrefix(String[] strs) {
    String prefix = strs[0];
    for (var i = 1; i < strs.length; ++i) {
      while (strs[i].indexOf(prefix) != 0) {
        prefix = prefix.substring(0, prefix.length() - 1);
        if (prefix.isEmpty()) return "";
      }
    }
    return prefix;
  }

  public static void main(String[] args) {
    var test = new Solution();
    String[] strs = {"flower", "flow", "flight"}; // TEST: => "fl"
    System.out.println(test.longestCommonPrefix(strs));
  }
}

File: Go/14.go
/**
 * Runtime: 2ms (Beats 63.36%)
 * Memory: 2.4MB (Beats 31.26%)
 */

package main

import "strings"

func longestCommonPrefix(strs []string) string {
	prefix := strs[0]
	for i := 1; i < len(strs); i++ {
		for strings.Index(strs[i], prefix) != 0 {
			prefix = prefix[:len(prefix)-1]
			if len(prefix) == 0 {
				return ""
			}
		}
	}
	return prefix
}
